packages/@lwc/synthetic-shadow/src/faux-shadow/events.ts

```diff
@@ -320,42 +143,14 @@ function detachDOMListener(elm: Element, type: string, wrappedListener: WrappedL
     }
 }
 
-function isValidEventForCustomElement(event: Event): boolean {
-    const target = eventTargetGetter.call(event);
-    const currentTarget = eventCurrentTargetGetter.call(event);
-    const { composed } = event as any;
-    return (
-        // it is composed, and we should always get it, or
-        composed === true ||
-        // it is dispatched onto the custom element directly, or
-        target === currentTarget ||
-        // it is coming from a slotted element
-        isChildNode(
-            getRootNodeHost(
-                target as Node /* because wrap on shadowRoot */,
-                GET_ROOT_NODE_CONFIG_FALSE
-            ) as Element,
-            currentTarget as Node
-        )
-    );
-}
-
 export function addCustomElementEventListener(
     elm: Element,
     type: string,
     listener: EventListener,
     _options?: boolean | AddEventListenerOptions
 ) {
-    if (process.env.NODE_ENV !== 'production') {
-        if (!isFunction(listener)) {
-            throw new TypeError(
-                `Invalid second argument for Element.addEventListener() in ${toString(
-                    elm
-                )} for event "${type}". Expected an EventListener but received ${listener}.`
-            );
-        }
-    }
```

Comment:
How come we are removing this guardrail?

Reply:
because it is happening one layer on top... in the patch to `addEventListener`, which wraps every listener, and calls this method with the wrapped (always existing) function.